WBooks/src/app/screens/Login/layout.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { ImageBackground, TextInput, Text } from 'react-native';
+import backgroundImage from '@assets/general/bc_inicio.png';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+import { USER_PLACE_HOLDER, PASSWORD_PLACE_HOLDER, LOG_IN } from './constants/texts';
+import styles from './styles';
+
+function LoginLayout({ updateUser, updatePassword, logInSuccessful }) {
+  return (
+    <ImageBackground style={styles.loginContainer} source={backgroundImage}>
+      <TextInput
+        style={[styles.textInput, styles.container]}
+        placeholder={USER_PLACE_HOLDER}
+        textAlign="center"
+        onChangeText={text => updateUser(text)}
+      />
+      <TextInput
```

Comment:
This `TextInout` is pretty similar to the upper one, try to abstract the repeated logic

Reply:
Do you think that i should make a component?

Reply:
I create a function to abstract the logic of the updates function, i made a partial function. it's something like this: update = param => text => this.setState({ [param]: text }); 
And it's used to the tupdatePassword and user, doing: updatePassword = update('password'), and then text => updatePassword(text). With the curry library is more simple to make partial functions.